components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/core/SessionManagementService.java

```diff
@@ -0,0 +1,180 @@
+import org.wso2.carbon.identity.api.user.common.error.ErrorResponse;
+import org.wso2.carbon.identity.api.user.session.common.constant.SessionManagementConstants;
+import org.wso2.carbon.identity.api.user.session.common.util.SessionManagementUtil;
+import org.wso2.carbon.identity.application.authentication.framework.exception.session.mgt
+        .SessionManagementClientException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.session.mgt.SessionManagementException;
+import org.wso2.carbon.identity.application.authentication.framework.model.UserSession;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.rest.api.user.session.v1.core.function.UserSessionToExternal;
+import org.wso2.carbon.identity.rest.api.user.session.v1.dto.SessionDTO;
+import org.wso2.carbon.identity.rest.api.user.session.v1.dto.SessionsDTO;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.user.common.Constants.ERROR_CODE_DELIMITER;
+import static org.wso2.carbon.identity.api.user.common.Util.resolveUserIdFromUser;
+import static org.wso2.carbon.identity.api.user.session.common.constant.SessionManagementConstants.ErrorMessage
+        .ERROR_CODE_FILTERING_NOT_IMPLEMENTED;
+import static org.wso2.carbon.identity.api.user.session.common.constant.SessionManagementConstants.ErrorMessage
+        .ERROR_CODE_PAGINATION_NOT_IMPLEMENTED;
+import static org.wso2.carbon.identity.api.user.session.common.constant.SessionManagementConstants.ErrorMessage
+        .ERROR_CODE_SORTING_NOT_IMPLEMENTED;
+
+/**
+ * Call internal osgi services to perform user session related operations.
+ */
+public class SessionManagementService {
+
+    private static final Log log = LogFactory.getLog(SessionManagementService.class);
+
+    /**
+     * Get all the active sessions of a given user.
+     *
+     * @param user   user
+     * @param limit  limit (optional)
+     * @param offset offset (optional)
+     * @param filter filter (optional)
+     * @param sort   sort (optional)
+     * @return SessionsDTO
+     */
+    public SessionsDTO getSessionsBySessionId(User user, Integer limit, Integer offset, String filter, String
+            sort) {
+
+        handleNotImplementedCapabilities(limit, offset, filter, sort);
+
+        List<UserSession> sessionsForUser;
+        SessionsDTO sessions = new SessionsDTO();
+        try {
+            String userId = resolveUserIdFromUser(user);
+            sessionsForUser = SessionManagementUtil.getUserSessionManagementService().getSessionsByUserId(userId);
+            sessions.setUserId(userId);
+            sessions.setSessions(buildSessionDTOs(sessionsForUser));
+            return sessions;
+
+        } catch (SessionManagementException e) {
+            log.error(e.getMessage());
+            handleSessionManagementException(e);
```

Comment:
return exception from the handleSessionManagementException method and throw from here, then no need return null

Reply:
done